Crash when enabling load_saved_merc_by_nickname game policy without a SavedGames folder present

```
2021-04-22T19:14:11 [ INFO] sgp/SGP.cc: Saved games directory:         '/home/kajak/.ja2/SavedGames-imp-quiz-honest-answers-from-russia-with-love'
2021-04-22T19:14:11 [ INFO] sgp/SGP.cc: ------------------------------------------------------------------------------
/.../
2021-04-22T19:14:11 [ INFO] game/GameLoop.cc: Version Label: Stracciatella v0.18.0-git+d30318a
2021-04-22T19:14:11 [ INFO] game/GameLoop.cc: Version #:     Build 04.12.02
2021-04-22T19:22:31 [ERROR] sgp/FileMan.cc: FileMan::openForWriting '/home/kajak/.ja2/SavedGames-imp-quiz-honest-answers-from-russia-with-love/mercprofile.nelel' true: File_open "/home/kajak/.ja2/SavedGames-imp-quiz-honest-answers-from-russia-with-love/mercprofile.nelel" 0x1a: No such file or directory (os error 2)
2021-04-22T19:22:31 [ERROR] sgp/FileMan.cc: FileMan::openForWriting '/home/kajak/.ja2/SavedGames-imp-quiz-honest-answers-from-russia-with-love/mercprofile.nelel' true: File_open "/home/kajak/.ja2/SavedGames-imp-quiz-honest-answers-from-russia-with-love/mercprofile.nelel" 0x1a: No such file or directory (os error 2)
2021-04-22T19:22:31 [ERROR] sgp/SGP.cc: Game has been terminated due to an unrecoverable error: FileMan::openForWriting failed (St13runtime_error)
2021-04-22T19:22:31 [ERROR] launcher/Launcher.cc: Command_execute "/tmp/.mount_ja2-st0TdIgN/usr/bin/ja2" []: terminated by a signal
```
The save dir wasn't created, so the failure makes sense.
But that's odd, since I'd expect a complaint from FileMan::createDir / Fs_createDir beforehand.

--- a/src/game/SaveLoadGame.cc
+++ b/src/game/SaveLoadGame.cc
@@ -195,8 +195,6 @@ static void SaveIMPPlayerProfiles();
 
 BOOLEAN SaveGame(UINT8 ubSaveGameID, const ST::string& gameDesc)
 {
-	if (gamepolicy(imp_load_saved_merc_by_nickname)) SaveIMPPlayerProfiles();
-
 	BOOLEAN	fPausedStateBeforeSaving    = gfGamePaused;
 	BOOLEAN	fLockPauseStateBeforeSaving = gfLockPauseState;
 
@@ -266,6 +264,9 @@ BOOLEAN SaveGame(UINT8 ubSaveGameID, const ST::string& gameDesc)
 		}
 
 		FileMan::createDir(GCM->getSavedGamesFolder().c_str());
+
+		// Save IMP merc(s)
+		if (gamepolicy(imp_load_saved_merc_by_nickname)) SaveIMPPlayerProfiles();
 
 		// Create the save game file
 		char savegame_name[512];

